src/CoreWCF.Primitives/src/CoreWCF/Description/DataContractSerializerOperationBehavior.cs

```diff
@@ -62,16 +62,41 @@ public bool IgnoreExtensionDataObject
 
         internal bool IgnoreExtensionDataObjectSetExplicit { get; set; }
 
+        public ISerializationSurrogateProvider SerializationSurrogateProvider { get; set; }
+
         public DataContractResolver DataContractResolver { get; set; }
 
         public virtual XmlObjectSerializer CreateSerializer(Type type, string name, string ns, IList<Type> knownTypes)
         {
-            return new DataContractSerializer(type, name, ns, knownTypes);
+            XmlDictionary dictionary = new XmlDictionary(2);
+            DataContractSerializerSettings settings = new DataContractSerializerSettings();
+            settings.RootName = dictionary.Add(name);
+            settings.RootNamespace = dictionary.Add(ns);
+            settings.KnownTypes = knownTypes;
+            settings.MaxItemsInObjectGraph = MaxItemsInObjectGraph;
+            settings.DataContractResolver = DataContractResolver;
+            DataContractSerializer dcs = new DataContractSerializer(type, settings);
+            if (SerializationSurrogateProvider != null)
```

Comment:
The null check is to prevent potential NullReferenceException issues due to the following .NET Framework implementation:
```
public static void SetSerializationSurrogateProvider(this DataContractSerializer serializer, ISerializationSurrogateProvider provider)
{
    IDataContractSurrogate dataContractSurrogate = new DataContractSerializerExtensions.SurrogateProviderAdapter(provider);
    serializer.SetDataContractSurrogate(dataContractSurrogate);
}
```
In this code, if the provider parameter is null, dataContractSurrogate is still created with a non-null instance. This leads to problems later when the code attempts to use the null provider, causing NullReferenceException (in netfx platform).